logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -762,9 +762,12 @@ class TestPipeline < LogStash::Pipeline
     end
     let(:dummyoutput) { ::LogStash::Outputs::DummyOutput.new({ "id" => dummy_output_id }) }
     let(:metric_store) { subject.metric.collector.snapshot_metric.metric_store }
+    let(:before_subject_hook) {}
     let(:pipeline_thread) do
       # subject has to be called for the first time outside the thread because it will create a race condition
       # with the subject.ready? call since subject is lazily initialized
+      # The hook is used to allow support for mocking,etc in tests
+      before_subject_hook
```

Comment:
Yeah, I don't like it much either, but other things I tried ran into a bit of a dead end... The combination of a Java static factory class for creating the DeadLetterQueueWriter and the fact that the subject is instantiated in a 'before :each' block made this a little tricky.
There are integration tests for these metrics, which make me a little happier, but if you have any other suggestions for better ways to test this, I'm open